code/ui/Hud.cs

```diff
@@ -23,54 +19,100 @@ public Hud()
                 return;
             }
 
-            Instance = this;
+            Current = this;
 
-            RootPanel.AddChild<ChatBox>();
-            RootPanel.AddChild<VoiceList>();
-            RootPanel.AddChild<GameTimer>();
-            RootPanel.AddChild<InfoFeed>();
-            RootPanel.AddChild<PostRoundMenu>();
-            RootPanel.AddChild<QuickShop>();
+            GeneralHudPanel = new GeneralHud(RootPanel);
+            AliveHudPanel = new AliveHud(RootPanel);
+        }
 
-            Scoreboard = RootPanel.AddChild<Scoreboard>();
+        [Event.Hotload]
+        public static void OnHotReloaded()
+        {
+            if (Host.IsClient)
+            {
+                Local.Hud?.Delete();
```

Comment:
What is this? :o I'd just use `Current.ClearChildren(true)`

Reply:
Then you don't have to destroy the whole hud and reinitialize a new object

Reply:
If we are going to do `new Hud();` it probably makes sense to just delete the entire object. Since on the next line we are constructing a brand new hud.